src/test/shell/bazel/unix_digest_hash_attribute_name_test.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+#
+# Copyright 2020 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# --- begin runfiles.bash initialization ---
+# Copy-pasted from Bazel's Bash runfiles library (tools/bash/runfiles/runfiles.bash).
+set -euo pipefail
+if [[ ! -d "${RUNFILES_DIR:-/dev/null}" && ! -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+    if [[ -f "$0.runfiles_manifest" ]]; then
+      export RUNFILES_MANIFEST_FILE="$0.runfiles_manifest"
+    elif [[ -f "$0.runfiles/MANIFEST" ]]; then
+      export RUNFILES_MANIFEST_FILE="$0.runfiles/MANIFEST"
+    elif [[ -f "$0.runfiles/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+      export RUNFILES_DIR="$0.runfiles"
+    fi
+fi
+if [[ -f "${RUNFILES_DIR:-/dev/null}/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+  source "${RUNFILES_DIR}/bazel_tools/tools/bash/runfiles/runfiles.bash"
+elif [[ -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  source "$(grep -m1 "^bazel_tools/tools/bash/runfiles/runfiles.bash " \
+            "$RUNFILES_MANIFEST_FILE" | cut -d ' ' -f 2-)"
+else
+  echo >&2 "ERROR: cannot find @bazel_tools//tools/bash/runfiles:runfiles.bash"
+  exit 1
+fi
+# --- end runfiles.bash initialization ---
+
+source "$(rlocation "io_bazel/src/test/shell/integration_test_setup.sh")" \
+  || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
+
+function test_xattr_operations_in_profile_log {
+  touch WORKSPACE
+  cat > BUILD << 'EOF'
+genrule(
+    name = "foo",
+    outs = ["foo.out"],
+    cmd = "touch $@",
+)
+EOF
+
+  bazel \
+      --unix_digest_hash_attribute_name=user.checksum.sha256 \
+      build \
+      --profile=profile_log \
+      --record_full_profiler_data \
+      //:foo || fail "Build failed"
+  grep -q "VFS xattr.*BUILD" profile_log || \
```

Comment:
It looks like tools for setting extended attributes aren't really standardised. macOS ships with xattr(1). On Linux there is supposedly setfattr(1), but it doesn't ship out of the box, even on mainstream distros (Ubuntu 18.04). This is why I've added some code to at least set the extended attribute on macOS.

With regards to checking whether the extended attribute got read, it doesn't look like the profile logs contain enough information to check for that. My hope was that the lines immediately following 'VFS xattr' would show that, but they don't.

Reply:
I think that UnixFileSystem logs a (poorly named) "md5" operation on a [manual digest](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/unix/UnixFileSystem.java#L417). Can you check that you don't have any such operations? You might need to do this in a more complex scenario, since Bazel doesn't digest files if their fast digest is missing unless it needs to. Such a need is likely from remote execution. Perhaps you would need to use remote_execution_test.sh or fork it for this.

As an aside, I'm actually not sure why we only profile in UnixFileSystem. Spelunking through the code history, @ulfjack made some changes in 2011(!) that might be relevant, but at this point, if you need to, you could lift the profiling up, I think. Not sure if that's necessary, though, since UnixFileSystem may be the relevant one for all the environments this would be used for.

Reply:
I've just added a test that does exactly this to remote_execution_with_xattr_test.sh.